lib/src/parser/proxy_protocol.rs

```diff
@@ -0,0 +1,226 @@
+use nom::IResult;
+use nom::IResult::*;
+use nom::{be_u8, be_u16, ErrorKind};
+
+use std::net::{Ipv4Addr, Ipv6Addr, SocketAddrV4, SocketAddrV6};
+use std::convert::AsMut;
+use std::convert::From;
+
+use network::protocol::proxy_protocol::header::*;
+
+pub const UNKNOWN_FAMILY: u32 = 1;
+pub const INCORRECT_VERSION: u32 = 2;
+
+const PROTOCOL_SIGNATURE_V2: [u8; 12] = [0x0D, 0x0A, 0x0D, 0x0A, 0x00, 0x0D, 0x0A, 0x51, 0x55, 0x49, 0x54, 0x0A];
+
+//NOTE: Only support the V2 for the moment
+pub fn parse(buf: &[u8]) -> IResult<&[u8], ProxyProtocolHeader> {
```

Comment:
this `tag!` should be part of the `do_parse` used in `parse_v2_header`. There's no need to pattern match manually here

Reply:
I'm facing with this issue: 
`expected array of 12 elements, found &[u8]` with 

``` rust
fn parse_v2_header(input: &[u8]) -> IResult<&[u8], HeaderV2> {
  do_parse!(
    input,
    signature: tag!(&PROTOCOL_SIGNATURE_V2) >>
    ver_and_cmd: read_and_verify_ver_and_command >>
    family: be_u8 >>
    len: be_u16 >>
    addr: apply!(parse_addr_v2, family) >>
    (
      HeaderV2 {
        signature,
        ver_and_cmd,
        family,
        len,
        addr
      }
    )
  )
}
```
How can I transform the part of the tag read to an array ?

Reply:
That doesn't work with `tag!(&PROTOCOL_SIGNATURE_V2[..])` steal the same issue and `[0..12]` too. Have you a parser in nom to change a slice to an array ? Maybe we can combine the tag with it.

Reply:
Done